plugins/outputs/mongodb/mongodb.go

```diff
@@ -229,5 +229,9 @@ func (s *MongoDB) Write(metrics []telegraf.Metric) error {
 }
 
 func init() {
-	outputs.Add("mongodb", func() telegraf.Output { return &MongoDB{} })
+	outputs.Add("mongodb", func() telegraf.Output {
+		return &MongoDB{
+			collections: make(map[string]bson.M),
```

Comment:
I don't think this change is necessary as we do not use `s.collections` before it is initialized in `getCollections` or do we?

Reply:
I wanted to avoid a potential situation where the map would not be initialized in `Connect`, but somehow someone would want to insert something into it (it doesn't look like this can happen with the current code, but if someone were to implement a `StartupError` here, then it could. 
Anyway, before this PR, the map was always initialized after exiting the `Connect` method.

Reply:
Actually if people implement this, they should make sure stuff is initialized. So please remove the code here.

Reply:
Done